### PR TITLE
fix cooling coil sizing issue for dry climates

### PR DESCRIPTION
**Pull request by EnergyArchmage**
Pull request overview
---------------------
address issue #6829 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Myoldmopar** [comment]
@EnergyArchmage is this one all ready to go?

**Comment by EnergyArchmage** [comment]
Not quite yet. still studying CI results.  Lots of diffs are just because the order of when things happen in water coil sizing is changed.  But there are more changes to sizing results than I expected.

**Comment by rraustad** [comment]
I single stepped this several times using 5ZoneAirCooled and Denver weather. Things appear to work as expected except that I can see a time when the user entered SA humidity ratio could be higher than the main coil inlet humdity ratio (e.g., 100 % OA). I tried to correct and got the same answer as this branch, likely because OA fraction is 21%. Instead of muddying the water here with those changes I reverted and left as is, with only minor other changes. A simple test case with 100% OA would serve as a new defect file for this other issue I noticed. I also wonder why temp couldn't have the same problem as seen here with humidity ratio, with user entered SAT or pre-cool T being higher that what is entering the coil. That is a matter of load profile but seems very unlikely.

Other than this, this correction seems reasonable to me.

**Comment by mjwitte** [comment]
Don't we already have code in place to lower the design leaving humrat and/or temp if either one is higher than entering?

**Comment by EnergyArchmage** [comment]
Yes, but it executes after capacity sizing. This rearranges existing code so that the altered conditions are used consistently.

On August 1, 2018 11:12:38 PM GMT+01:00, "Michael J. Witte" <notifications@github.com> wrote:
>Don't we already have code in place to lower the design leaving humrat
>and/or temp if either one is higher than entering?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/NREL/EnergyPlus/pull/6830#issuecomment-409742042

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: src/EnergyPlus/ReportSizingManager.cc [hunk 1]
```diff
@@ -2444,8 +2445,10 @@ namespace ReportSizingManager {
                     } else {
                         AutosizeDes = FinalSysSizing(CurSysNum).CoolSupHumRat;
                     }
-                    if (AutosizeDes > DataDesInletAirHumRat && DataCapacityUsedForSizing > 0.0 &&
-                        UtilityRoutines::SameString(CompType, "COIL:COOLING:WATER")) { // flow here is water vol flow rate
+                    if (AutosizeDes > DataDesInletAirHumRat &&
+                        UtilityRoutines::SameString(CompType,
+                                                    "COIL:COOLING:WATER")) { // flow here is water vol flow rate, take out DataCapacityUsedForSizing >
```
Comment (rraustad) [inline src/EnergyPlus/ReportSizingManager.cc]: Shouldn't these warnings also apply to the detailed cooling coil model?

Comment (rraustad) [reply src/EnergyPlus/ReportSizingManager.cc]: Or any cooling coil for that matter.

Comment (rraustad) [reply src/EnergyPlus/ReportSizingManager.cc]: Well, only water coils have an inlet water temp so OK, simple and detailed then.

**Comment by rraustad** [comment]
The diff's I have reviewed so far show that this change has corrected the coil inlet conditions used to size water coils. I wonder if it would have been better to correct the pre-cool humrat and/or T at the end of sizing (e.g., pre-cool humrat = min(pre-cool humrat, OA humrat)), but then that would corrupt the user inputs (which we override anyway with this change) and it's probably better to retain user inputs for unforeseen purposes. I'll take another pass at the diff's when CI finishes.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by EnergyArchmage** [comment]
Those code fixes related to water-cooled vs air-cooled DX had crept in here but are really off topic for the dry-water coil issue.  So I backed them out.  That is this issue: https://github.com/NREL/EnergyPlus/issues/6376 and (presumably) should have its own pull request.  Not sure if anyone is working on that, I could do it.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 10 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by mjwitte** [comment]
@EnergyArchmage @rraustad Is this complete, other than merging in develop and resolving conflicts?

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 9 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@EnergyArchmage @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 9 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@EnergyArchmage it has been 7 days since this pull request was last updated.

**Comment by rraustad** [comment]
Moved changes to #7196. Delete this branch when that PR is complete.

**Comment by Myoldmopar** [comment]
I'm going to go ahead and close this one out and reassign #7196 for the current release.  Feel free to undo either of these changes if needed.

**Pull request closed without merging**